### DISCUSSION TITLE
[Javascript Testing] Add tests for Filter component

### DISCUSSION THREAD

**Discussion by FeminismIsAwesome** [Terrastories/terrastories#744] [Temporarily Archived Tickets]
### Background:
We would like more confidence, especially with desires to update the UI and add new map-facing features, that the javascript is rendering as expected. This is a good place for those new to the project to start getting familiar with the javascript end of things.

Let's add tests for the Filter.jsx component!

### Some suggestions for tests:

Adding an enzyme snapshot test (https://medium.com/codeclan/testing-react-with-jest-and-enzyme-20505fec4675)
Asserting that different if-block renders render with the right data (different filterItem)
Testing interactions, like "when this clicks, this state changes" 

### Acceptance:

Running `docker-compose exec web yarn test` with the new tests does not break the javascript test suite.

**Comment by leosoaivan** [comment]
May I take this one?

**Comment by FeminismIsAwesome** [comment]
@leosoaivan yes i think nobody else is working on it!

**Comment by leosoaivan** [comment]
@FeminismIsAwesome having a tough time with this one.

Snapshot testing is fine, but I'm finding that the functionality is a bit difficult to test without fully mocking out the functions in App.js that handle the change in Filter's props, namely `handleFilterCategoryChange` and `handleFilterItemChange`. I think it's wiser _not_ to attempt to mock out these functions, as that could lead to false positives in the future if these two functions ever change.

Since the handling of Filter's props has been hoisted up to `App.,js`, it may make more sense to test Filter's functionality in tests there.

Additionally, after further looking, it would appear that the fix for https://github.com/Terrastories/terrastories/issues/564 would have to be in App.js as well, since that is where `itemOptions` is managed.

**Comment by leosoaivan** [comment]
Any updates or thoughts on this?
